cds/modules/records/tasks.py

```diff
@@ -403,18 +403,14 @@ def _format_report(report):
         return '\n'.join(lines)
 
     cache = current_cache.get('task_missing_subformats:details') or {}
-    if 'last_run' not in cache:
-        # Save the last week date to have a default value the first time
-        cache['last_run'] = datetime.utcnow() - timedelta(days=7)
+    if 'end_date' not in cache:
+        # Set the end date to 7 days ago
+        cache['end_date'] = datetime.utcnow() - timedelta(days=7)
 
-    query = _filter_by_last_created(
+    record_uuids = _filter_by_last_created(
                 _get_all_records_with_bucket(),
-                start_date or cache['last_run'],
-                end_date)
-
-    record_uuids = query \
-        .group_by(RecordMetadata.id) \
-        .with_entities(RecordMetadata.id)
+                start_date,
```

Comment:
Here we check all records until one week ago, right?

Reply:
Yes, so if you pass an end date and not a start date then it will check for everything until one week ago. If you pass a start date then it will check from the start date until present - if you pass both than it will take the interval.